android/app/src/main/java/com/zulipmobile/notifications/GCMPushNotifications.java

```diff
@@ -90,6 +90,10 @@ public void onOpened() {
         // First, get a builder initialized with defaults from the core class.
         final Notification.Builder builder = super.getNotificationBuilder(intent);
 
+        if (!getProps().getEvent().equals("message")) {
+            return builder;
```

Comment:
I'm not sure what this is doing. Is this to handle something that starts happening in the dismiss-event case, because we no longer return `null` there?

Reply:
Basically to save some processing of the events as anyways the notification will be dumped afterward in the onReceived call.
So this avoids the modification of the builder, and gets dumped later on.

This is the stacktrace when a remove notification is recieved on the app now.
```
08-09 17:57:28.136 5819-6633/com.zulipmobile.debug V/ReactNativeNotifs: GCM message handling aborted
    com.wix.reactnativenotifications.core.notification.IPushNotification$InvalidNotificationException: Cannot handle this notification.
        at com.zulipmobile.notifications.GCMPushNotifications.onReceived(GCMPushNotifications.java:208)
        at com.wix.reactnativenotifications.gcm.GcmMessageHandlerService.onMessageReceived(GcmMessageHandlerService.java:20)
        at com.google.android.gms.gcm.GcmListenerService.handleIntent(Unknown Source:202)
        at com.google.firebase.iid.zzg.run(Unknown Source:26)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
        at java.lang.Thread.run(Thread.java:764)
```

Reply:
I've now studied the surrounding code quite a bit in the course of writing my own fix for this issue -- see linked commits in my main-thread comment.

Can you explain more about when you believe this `if` will be true and this bit of code will have an effect?

As far as I can see what's happening in the code, this `if` will never be true.
* The one place `getNotificationBuilder` is called is inside the base class's `buildNotification`; which is called only in its `postNotification`; which is called only in its `onPostRequest` (for local notifications, which we never do) and `onReceived`.
* With this commit's change to our own `onReceived`, the base class's `onReceived` is only called if this same condition is false.
* So this code is never reached if this condition is true (i.e. if `event` is not `message`.)